examples/image_segmentation/android/app/src/main/java/com/google/mediapipe/examples/imagesegmenter/ImageSegmenterHelper.kt

```diff
@@ -0,0 +1,263 @@
+            bitmapBuffer.height,
+            matrix,
+            true
+        )
+
+        val mpImage = BitmapImageBuilder(rotatedBitmap).build()
+
+        imagesegmenter?.segmentAsync(mpImage, frameTime)
+    }
+
+    // Runs image segmentation on single image and
+    // returns the results asynchronously to the caller.
+    fun segmentImageFile(mpImage: MPImage) {
+        if (runningMode != RunningMode.IMAGE) {
+            throw IllegalArgumentException(
+                "Attempting to call segmentImageFile" + " while not using RunningMode.IMAGE"
+            )
+        }
+        imagesegmenter?.segment(mpImage)
+    }
+
+    // Runs image segmentation on each video frame and
+    // returns the results asynchronously to the caller.
+    @kotlin.jvm.Throws(Exception::class)
+    fun segmentVideoFile(mpImage: MPImage) {
+        if (runningMode != RunningMode.VIDEO) {
+            throw IllegalArgumentException(
+                "Attempting to call segmentVideoFile" + " while not using RunningMode.VIDEO"
+            )
+        }
+        imagesegmenter?.segmentForVideo(mpImage, SystemClock.uptimeMillis())
+    }
+
+    // MPImage isn't necessary for this example, but the listener requires it
+    private fun returnSegmentationResult(
+        result: ImageSegmenterResult, image: MPImage
+    ) {
+        val finishTimeMs = SystemClock.uptimeMillis()
+
+        val inferenceTime = finishTimeMs - result.timestampMs()
+
+        // We only need the first mask for this sample because we are using
+        // the OutputType CATEGORY_MASK, which only provides a single mask.
+        val mpImage = result.segmentations().first()
+
+        imageSegmenterListener?.onResults(
+            ResultBundle(
+                ByteBufferExtractor.extract(mpImage),
+                mpImage.width,
+                mpImage.height,
+                inferenceTime
+            )
+        )
+    }
+
+    // Return errors thrown during segmentation to this
+    // ImageSegmenterHelper's caller
+    private fun returnSegmentationHelperError(error: RuntimeException) {
+        imageSegmenterListener?.onError(
+            error.message ?: "An unknown error has occurred"
+        )
+    }
+
+    // Wraps results from inference, the time it takes for inference to be
+    // performed.
+    data class ResultBundle(
+        val results: ByteBuffer,
+        val width: Int,
+        val height: Int,
+        val inferenceTime: Long,
+    )
+
+    companion object {
+        const val DELEGATE_CPU = 0
+        const val DELEGATE_GPU = 1
+        const val OTHER_ERROR = 0
+        const val GPU_ERROR = 1
+        const val MODEL_PATH = "deeplabv3.tflite"
+
+        private const val TAG = "ImageSegmenterHelper"
+
+        val labelColors = listOf(
+            Pair("background", -16777216),
+            Pair("aeroplane", -8388608),
+            Pair("bicycle", -16744448),
+            Pair("bird", -8355840),
+            Pair("boat", -16777088),
```

Comment:
We could probably switch this to just a list that we reference rather than by label, yeah? I do like having the names of the labels somewhere though - could maybe put them in a comment at the top of ImageSegmenterHelper